daisy/persistence/mongodb_graph_provider.py

```diff
@@ -321,8 +335,20 @@ def read_edges(self, roi, nodes=None, attr_filter=None, read_attrs=None):
                 i_b = i*query_size
                 i_e = min((i + 1)*query_size, len(node_ids))
                 assert i_b < len(node_ids)
-                endpoint_query = {self.endpoint_names[0]:
-                                  {'$in': node_ids[i_b:i_e]}}
+                if edge_inclusion == 'u':
+                    endpoint_query = {u: {'$in': node_ids[i_b:i_e]}}
+                elif edge_inclusion == 'v':
+                    endpoint_query = {v: {'$in': node_ids[i_b:i_e]}}
+                elif edge_inclusion == 'both':
+                    raise NotImplementedError(
```

Comment:
maybe this should be implemented before merging

Reply:
You can't just make an `$and` query since the node ids are split up into arbitrary queries, and there is no guarantee that both endpoints of an edge will be in the same query. You might just have to intersect the results from querying each endpoint independently. Or, retrieve all results for one endpoint and locally check to see if the retrieved edges have endpoints in the node_id set. Both seem inefficient, but one might be the best we can do with a non-relational database.